supplemental_notebooks/2022-09-19_muddiest_points.Rmd

```diff
@@ -0,0 +1,218 @@
+  "PSMC5",       8.94,            3.12, 
+  "IMPA1",      -1.10,            0.506,
+  "FABP7",      -5.15,            1.70, 
+  "SCARB2",     -3.28,            4.37, 
+  "FRMD6",       2.33,            0.929,
+  "SKA2",        6.35,            5.10 
+)
+
+# Print the data:
+gene_data
+```
+
+
+
+Now, let's make a plot where we also label points whose log fold change is greater than 5 (for the purposes of this example, we won't worry too much about significant p-values!).
+A helpful R package for labeling points, while ensuring those labels do not overlap one another, is [`ggrepel`](https://ggrepel.slowkow.com/). 
+We'll use the function `geom_text_repel()` from this package to create our labels.
+If you are using this package for the first time, you'll have to install it with `install.packages()`:
+```{r, eval=FALSE}
+install.packages("ggrepel")
+```
+
+
+First, let's plot a regular volcano plot without point labels, as we did during the workshop.
+
+```{r}
+# Load libraries needed for plotting:
+library(ggplot2)
+library(ggrepel)
+
+ggplot(
+  gene_data, 
+  aes(
+    x = log_fold_change,
+    y = neg_log10_p,
+    color = log_fold_change
+  )
+) + 
+  geom_point() + 
+  theme_bw() # add bw theme
+```
+
+Now, let's add some labels with the `ggrepel` function `geom_text_repel()`, again focusing only on points with >5 log fold change.
+
+
+```{r}
+ggplot(
+  gene_data, 
+  aes(
+    x = log_fold_change,
+    y = neg_log10_p,
+    color = log_fold_change
+  )
+) + 
+  geom_point() +
+  # This layer adds labels:
+  geom_text_repel(
+    # First, we want only a subset of data,which we can specify with the `data` argument. 
+    # We can use dplyr::filter for this!
+    data = dplyr::filter(gene_data, log_fold_change > 5),
+    # Next, we set up the label to map to the `gene_symbol` column in our data
+    aes(label = gene_symbol)
+  ) + 
+  theme_bw() 
+```
+
+If you prefer that all labels are the same color rather than matching the color of their points, you can specify by adding a `color` argument to `geom_text_repel`:
+
+
+```{r}
+ggplot(
+  gene_data, 
+  aes(
+    x = log_fold_change,
+    y = neg_log10_p,
+    color = log_fold_change
+  )
+) + 
+  geom_point() +
+  # This layer adds labels:
+  geom_text_repel(
+    # First, we want only a subset of data,which we can specify with the `data` argument. 
+    # We can use dplyr::filter for this!
+    data = dplyr::filter(gene_data, log_fold_change > 5),
+    # Next, we set up the label to map to the `gene_symbol` column in our data
+    aes(label = gene_symbol),
+    # Specify the same color (here, black) for all points
+    color = "black"
+  ) + 
+  theme_bw() 
+```
+
+
+We'll also point out this super useful Bionconductor package for volcano plots called [`EnhancedVolcano`](https://bioconductor.org/packages/release/bioc/vignettes/EnhancedVolcano/inst/doc/EnhancedVolcano.html), which can automatically take care of a lot of labeling and styling for you, without needing to use `ggplot2` directly.
```

Comment:
I think it is worth adding a note that the plots are ggplots, but you might word it differently.
```suggestion
We'll also point out this super useful Bionconductor package for volcano plots called [`EnhancedVolcano`](https://bioconductor.org/packages/release/bioc/vignettes/EnhancedVolcano/inst/doc/EnhancedVolcano.html), which can automatically take care of a lot of labeling and styling for you, without needing to use `ggplot2` directly.
Handily though, the plots it creates _are_ `ggplot2` plots, so you can customize them further with the same layer syntax that you have been using!
```